sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -164,9 +207,13 @@ public String getId() {
         if (options == null) {
             options = new CosmosContainerRequestOptions();
         }
-        return database.getDocClientWrapper()
-                   .replaceCollection(ModelBridgeInternal.getV2Collection(containerProperties), ModelBridgeInternal.toRequestOptions(options))
-                   .map(response -> ModelBridgeInternal.createCosmosAsyncContainerResponse(response, database)).single();
+
+        if(!database.getClient().getTracerProvider().isEnabled()) {
```

Comment:
Seeing this is used frequently throughout the class, can this be extracted to somewhere common?

Reply:
Eventually this check will go away when we will go via withContext all the time.